content/moderation/moderation-overview.md

```diff
@@ -0,0 +1,324 @@
+
+{{% /notice %}}
+
+### Warnings
+
+Warning a user adds an entry to the user's warning log on the server, and optionally appears in the mod log as well.
+This can be used to better inform future moderation actions based on the user's past behavior. Additionally, it sends a
+DM to the user with the warning reason, which can be used to dissuade a user from continuing harmful actions without
+needing to remove their access to your server.
+
+#### Message Logs
+
+You may optionally allow YAGPDB to create a [message log](/moderation/logging) of the last 100 messages from all users
+in the channel a warning was issued in. This can help provide context for past warnings and further inform future mod
+actions. This is enabled in the warnings tab on the control panel.
+
+#### Warning Mod Log
+
+You may optionally send warnings to the mod log, this is enabled in the warnings tab on the control panel.
+
+#### Managing Warnings
+
+[Warn Command Syntax](/commands#warn)
+
+[Warnings Command Syntax](/commands#warnings) - Lists warnings of a user.
+
+[EditWarning Command Syntax](/commands#editwarning) - Retroactively edits a warning.
+
+[DelWarning Command Syntax](/commands#delwarning) - Deletes a warning.
+
+[ClearWarnings Command Syntax](/commands#clearwarnings) - Clears the warnings of a user.
+
+[TopWarnings Command Syntax](/commands#topwarnings) - Shows ranked list of warnings on the server, allowing you to turn
+misbehavior into a fun minigame for your server members.
+
+{{% notice style="tip" title="Taking Action on Warn Count" %}}
+
+Some users prescribe an "x warnings => mute/kick/ban" rule for their servers. To have YAGPDB automatically take action
+based on warning count, you can take advantage of the template scripting within the warning DM.
+
+Example:
+
+```go
+{{/* Config Area Starts */}}
+
+{{$count := 4 }} {{/* No of warnings at which Certain Action is to be taken ( eg: for action to take place at 4 warnings set count as 4 ) */}}
+{{$some_channel_ID := 0}} {{/* replace with channel id of any random channel which is used for command execution when a default channel of warn doesn't exist */}}
+
+{{/* Config Area Ends */}}
+
+{{define "punish_check"}}
+{{$count := toInt .TemplateArgs}}
+{{$warns_count := toInt (reFind `\d+` (execAdmin "Warnings" .User.ID).Description)}}
+
+{{/* If warning count is above a threshold value take action */}}
+{{if ge $warns_count $count}}
+
+{{/* Add required code for action to be taken when warning count exceeds the threshold */}}
+Add Code Here
+
+{{/* Code to reset warnings for the user. Delete if you do not wish to reset the warnings */}}
+{{$silent := execAdmin "ClearWarnings" .User.ID}}
+
+{{/*End statement for if. Do not Remove if you don't know what you're doing*/}}
+{{end}}
+
+{{end}}
+
+{{if .Channel}}
+     {{$s := sendTemplate .Channel.ID "punish_check" $count}}
+{{else}}
+     {{$s := sendTemplate $some_channel_ID "punish_check" $count}}
+{{end}}
```

Comment:
If this example is to be part of the official documentation, we should clean it up a little. (Note: untested.)
```suggestion
{{ $threshold := 4 }} {{/* Number of warnings at which action is to be taken (eg: for action to take place at 4 warnings set threshold to 4) */}}

{{ define "punish_check" }}
	{{ $threshold := toInt .TemplateArgs }}
	{{ $warnCount := toInt (reFind `\d+` (execAdmin "warnings" .User).Description) }}

	{{if ge $warns_count $count}}
		{{/* Add required code for action to be taken when warning count exceeds the threshold. */}}
		Add code here

		{{/* Reset warnings for the user. Delete if you do not wish to reset the warnings. */}}
		{{ $silent := execAdmin "clearwarnings" .User }}
	{{ end }}
{{ end }}

{{ if .Channel }}
     {{ $s := sendTemplate .Channel.ID "punish_check" $count }}
{{ else }}
	{{/* no context channel; use an arbitrary one */}}
     {{ $s := sendTemplate (index .Guild.Channels 0).ID "punish_check" $count }}
{{ end }}
```

Reply:
`$count` was only changed to `$threshold` in the definitions, and this should use `getWarnings` instead of parsing an embed from `execAdmin`.